pegasus/src/database.rb

```diff
@@ -63,14 +63,28 @@ def find_with_grade_level(gradelevel)
 
   # return the first tutorial with a matching code
   def find_with_code(code)
-    by_code = CDO.cache.fetch("Tutorials/#{@table}/by_code") {@contents.index_by {|row| row[:code]}}
-    by_code[code]
+    ids_by_code = CDO.cache.fetch("CdoTutorials/#{@table}/ids_by_code") do
+      @contents.index_by {|row| row[:code]}.transform_values {|obj| obj[:id]}
+    end
```

Comment:
There's a part of me that wants to pull these out into their own helper methods, but there's also a part of me that thinks that's a bit excessive and that what you have here is just fine.

Reply:
Personally, I don't think it would be cleaner in a helper method. Are there any comments or such that would be helpful?